src/main/java/com/hazelcast/commandline/HazelcastMember.java

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2020 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OF ANY KIND, either express or implied. See the License for the
+ * specific language governing permissions and limitations under the License.
+ */
+
+package com.hazelcast.commandline;
+
+import com.hazelcast.config.Config;
+import com.hazelcast.config.ConfigStream;
+import com.hazelcast.config.FileSystemXmlConfig;
+import com.hazelcast.config.FileSystemYamlConfig;
+import com.hazelcast.config.InterfacesConfig;
+import com.hazelcast.config.InvalidConfigurationException;
+import com.hazelcast.core.Hazelcast;
+import com.hazelcast.internal.config.MemberXmlConfigRootTagRecognizer;
+import com.hazelcast.internal.config.MemberYamlConfigRootTagRecognizer;
+
+import java.io.FileInputStream;
+
+import static com.hazelcast.internal.util.StringUtil.isNullOrEmpty;
+
+/**
+ * Class for starting new Hazelcast members
+ */
+public final class HazelcastMember {
```

Comment:
I wonder if it's not confusing that you embed Hazelcast Member, but use MC from the separate JAR. Would it maybe make sense to embed MC as well?

Reply:
It was my initial intention, however the Hazelcast dependency on MC clashes with the version that is used in the CLI. It means, the CLI should always be dependent on MC's Hazelcast version. Additionally, MC is never designed as a dependency library, thus there are many other issues need to be resolved.

Reply:
>  however the Hazelcast dependency on MC clashes with the version that is used in the CLI

How come? Can't we ensure that we always use matching versions?

Reply:
Nope, MC pickes a Hazelcast version (which can be older than the latest) and adds it as a dependency. But it can be still compatible with the latest version. Also, if we use MC as a dependency, and would like to upgrade Hazelcast's version then we'll need to wait for MC to upgrade it.